From bba1b0085f2a2c83839647fee48628ee054aceb7 Mon Sep 17 00:00:00 2001
From: "Dmitry V. Levin" <ldv@altlinux.org>
Date: Sun, 15 Jul 2018 22:14:29 +0000
Subject: [PATCH] tests: fix build with fresh glibc

Workaround additional incompatibility between <sys/stat.h> and
<asm/stat.h> introduced by glibc-2.27.9000-566-gfd70af4.

* tests/xstatx.c (statx, statx_timestamp): Redefine before the inclusion
of <sys/stat.h> to avoid conflicts between definitions provided by the
linux kernel headers and the GNU libc.

Upstream-Status: Backport [https://gitlab.com/strace/strace/commit/3dbffb7]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---

 tests/xstatx.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tests/xstatx.c b/tests/xstatx.c
index 7fb9326..441f4a8 100644
--- a/tests/xstatx.c
+++ b/tests/xstatx.c
@@ -64,10 +64,14 @@ typedef off_t libc_off_t;
 
 # define stat libc_stat
 # define stat64 libc_stat64
+# define statx libc_statx
+# define statx_timestamp libc_statx_timestamp
 # include <fcntl.h>
 # include <sys/stat.h>
-# undef stat
+# undef statx_timestamp
+# undef statx
 # undef stat64
+# undef stat
 
 # undef st_atime
 # undef st_mtime
-- 
2.18.0

